Sources/Widgets/Widgets3D/AngleWidget/behavior.js

```diff
@@ -60,7 +60,7 @@ export default function widgetBehavior(publicAPI, model) {
       newHandle.setColor(moveHandle.getColor());
       newHandle.setScale1(moveHandle.getScale1());
       newHandle.setManipulator(manipulator);
-    } else {
+    } else if (model.dragable) {
```

Comment:
if not dragable, you still call `StartInteractionEvent` but you won't call `EndInteractionEvent`

Reply:
I agree we need to fix that, but this suggest we should change `handleLeftButtonRelease` method.
I found weird we always return `macro.VOID` or `macro.EVENT_ABORT` in `handleLeftButtonPress` and `handleMouseMove` but not in this one.

I suggest to standardize the `handleLeftButtonRelease` method using the template below (to adapt depending of the widgets)
```js
publicAPI.handleLeftButtonRelease = () => {
  // Discard useless events
  if (
    !model.activeState ||
    !model.activeState.getActive() ||
    !model.pickable
  ) {
    return macro.VOID;
  }
  
  if (model.pickable && model.dragable && isDragging) {
    // Finish drag
    model._apiSpecificRenderWindow.setCursor('pointer');
    model.widgetState.deactivate();
    model._interactor.cancelAnimation(publicAPI);
    isDragging = false;
  } else if (model.activeState !== model.widgetState.getMoveHandle()) {
    model.widgetState.deactivate();
  }
  
  if (
    (model.activeState && !model.activeState.getActive())
  ) {
    model._widgetManager.enablePicking();
    model._interactor.render();
  }
  
  publicAPI.invokeEndInteractionEvent();
  return macro.EVENT_ABORT;
}
```
WDYT?

Reply:
I wouldn't even trigger Start/End"Interaction"event if the line is not draggable.

Reply:
But if the widget is pickable but not dragable? We may want to trigger a start/end event so that the user can know when the widget is "clicked/selected"?

Reply:
fair enough, that's good with me then.